src/Compilers/Test/Core/Mocks/Silverlight.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Roslyn.Test.Utilities;
+
+/// <summary>
+/// The assemblies produced here are designed to mimic the public key token structure of 
+/// silverlight references. This often presents challenges to the compiler because it has
+/// to know that two mscorlib with different public key tokens need to be treated as the 
+/// identicial. The assemblies produced here have the same identity of those that come
+/// from silverlight but without necessarily the same type contents.
+/// </summary>
+public static class Silverlight
+{
+    private static readonly Lazy<(byte[], byte[])> s_tuple = new Lazy<(byte[], byte[])>(
+        () => BuildImages(),
+        LazyThreadSafetyMode.PublicationOnly);
+
+    public static byte[] Mscorlib => s_tuple.Value.Item1;
+
+    public static byte[] System => s_tuple.Value.Item2;
+
+    private static (byte[], byte[]) BuildImages()
+    {
+        const string corlibExtraCode = """
+            using System;
+            using System.Reflection;
```

Comment:
Nit: these usings look unnecessary. Maybe adding VerifyDiagnostics to the compilations we create here would catch that?